c_glib/arrow-glib/compute.cpp

```diff
@@ -7091,6 +7093,157 @@ garrow_extract_regex_options_new(void)
   return GARROW_EXTRACT_REGEX_OPTIONS(options);
 }
 
+enum {
+  PROP_ZERO_FILL_OPTIONS_WIDTH = 1,
+  PROP_ZERO_FILL_OPTIONS_PADDING,
+};
+
+typedef struct _GArrowZeroFillOptionsPrivate GArrowZeroFillOptionsPrivate;
+struct _GArrowZeroFillOptionsPrivate
+{
+  gchar *padding;
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowZeroFillOptions,
+                           garrow_zero_fill_options,
+                           GARROW_TYPE_FUNCTION_OPTIONS)
+
+#define GARROW_ZERO_FILL_OPTIONS_GET_PRIVATE(object)                                     \
+  static_cast<GArrowZeroFillOptionsPrivate *>(                                           \
+    garrow_zero_fill_options_get_instance_private(GARROW_ZERO_FILL_OPTIONS(object)))
+
+static void
+garrow_zero_fill_options_dispose(GObject *object)
+{
+  auto priv = GARROW_ZERO_FILL_OPTIONS_GET_PRIVATE(object);
+  if (priv->padding) {
+    g_free(priv->padding);
+    priv->padding = nullptr;
+  }
+  G_OBJECT_CLASS(garrow_zero_fill_options_parent_class)->dispose(object);
+}
+
+static void
+garrow_zero_fill_options_set_property(GObject *object,
+                                      guint prop_id,
+                                      const GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto options = garrow_zero_fill_options_get_raw(GARROW_ZERO_FILL_OPTIONS(object));
+  auto priv = GARROW_ZERO_FILL_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_ZERO_FILL_OPTIONS_WIDTH:
+    options->width = g_value_get_int64(value);
+    break;
+  case PROP_ZERO_FILL_OPTIONS_PADDING:
+    {
+      const gchar *padding = g_value_get_string(value);
+      if (priv->padding) {
+        g_free(priv->padding);
+      }
+      priv->padding = g_strdup(padding);
+      options->padding = padding ? padding : "";
+    }
```

Comment:
Can we simplify this?

```suggestion
    options->padding =  g_value_get_string(value);
```